services/sr_asset_assign.py

```diff
@@ -0,0 +1,267 @@
+
+
+    object_id = config["object"]
+    status_field = config["assign_status_field_id"]
+    connected_field = config["connection_field_keys"][args.asset]
+    output_keys = ["id", config["asset_type_field_id"], connected_field]
+    layer = ASSET_CONFIG[args.asset]
+
+    for record in data:
+        if (
+            record[config["x_field"]] and record[config["y_field"]]
+        ):  # ignore records that have a null location data
+            point = [
+                record[config["x_field"]],
+                record[config["y_field"]],
+            ]
+            params = get_params(layer["layer"], point, token)
+            res = point_in_poly(
+                layer["layer"]["service_name"], layer["layer"]["layer_id"], params
+            )
+            # we have to manually check for response errors. The API returns `200` regardless
+            if res.get("error"):
+                raise Exception(str(res))
+            if not res["features"]:
+                # Only need to send the status no_asset_found
+                record[status_field] = "no_asset_found"
+                record[config["asset_type_field_id"]] = ""
+                record[connected_field] = ""
+                logger.info(f"No Assets found for ID: {record['id']}.")
+            elif len(res["features"]) != 1:
+                # Ignore records with multiple assets found
+                logger.info(
+                    f"Multiple Assets found for ID: {record['id']}, skipping updating this record."
+                )
+                continue
+            if len(res["features"]) == 1:
+                logger.info(f"One Asset found for ID: {record['id']}.")
+                record = handle_features(layer, record, connected_field, res)
+                record[config["asset_type_field_id"]] = layer["display_name"]
+
+            # Updating a record in Knack
+            record = {key: record[key] for key in output_keys}
+            # Add modified date field for every record:
+            record[modified_date_field] = local_timestamp()
+            try:
+                knackpy.api.record(
+                    app_id=APP_ID,
+                    api_key=API_KEY,
+                    obj=object_id,
+                    method="update",
+                    data=record,
+                )
+            except Exception as e:
+                logger.info(e.response.text)
+
+
+if __name__ == "__main__":
+    # CLI arguments definition
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument(
+        "-a",
+        "--app-name",
+        type=str,
+        help="str: Name of the Knack App in knack.py config file",
+    )
+
+    parser.add_argument(
+        "-c",
+        "--container",
+        type=str,
+        help="str: AKA API view that was created for downloading the location data",
+    )
+
+    parser.add_argument(
+        "-s",
+        "--asset",
+        type=str,
+        help="str: Name of the asset to search for potential matches",
```

Comment:
@Charlie-Henry it would be a nice touch to add an array of [`choices`](https://docs.python.org/3/library/argparse.html#choices) here to make it clear what assets names are currently supported/valid 🚦